spec/features/principal_creates_adviser_spec.rb

```diff
@@ -32,6 +32,42 @@
     and_i_can_try_another_adviser_reference_number
   end
 
+  scenario 'Matching an Adviser' do
+    given_i_have_created_a_firm
+    and_the_adviser_is_matched
+    when_i_request_the_advisers_name
+    then_the_endpoint_responds_ok
+    and_i_am_given_the_advisers_name
+  end
+
+  scenario 'Attempting to match a non-existent Adviser' do
+    given_i_have_created_a_firm
+    when_i_request_a_non_existent_adviser
+    then_the_endpoint_responds_404
+  end
+
+
```

Comment:
Yes, we've done that in this project's feature specs just to add some clear separation between the scenarios and their related steps.